R/pickle.R

```diff
@@ -11,7 +11,8 @@
 py_save_object <- function(object, filename, pickle = "pickle") {
   builtins <- import_builtins()
   pickle <- import(pickle)
-  handle <- builtins$open(filename, "wb")
+  norm_filename <- normalizePath(filename, mustWork = FALSE)
```

Comment:
@kevinushey Would it be better to use `path.expand` here? (not sure if there are other side effects of `normalizePath` that we'd want to avoid?)

Reply:
I would also lean towards using `path.expand()` here as well and let any path normalization (beyond tilde expansion) that needs to happen, happen on the Python side.

Reply:
Well, Python doesn't do tilde expansion, it has to be done manually with `os.path.expanduser`. On the other hand, R does tilde expansion, for instance, any `file.*` or `dir.*` base function.
So I thought, as py_load/save_object are R functions, it would be nice to have tilde expansion automatically done.
However, I agree with any suggestion @kevinushey might have, his knowledge of Python must be highly greater than mine 😅.

I didn't know about the `path.expand` function, `normalizePath` uses it:
```r
> path.expand
function (path) 
.Internal(path.expand(path))
<bytecode: 0x55e51aff80a0>
<environment: namespace:base>
> normalizePath
function (path, winslash = "\\", mustWork = NA) 
.Internal(normalizePath(path.expand(path), winslash, mustWork))
<bytecode: 0x55e51affb408>
<environment: namespace:base>
```
So, thanks for the knowledge! ☺️

Reply:
The main thing I was thinking about was Windows. Python does understand the different forms of Windows paths (forward slash, backslash, short path names) and may understand UNC paths as well. If we perform normalization on the R side we risk losing some user intent or getting some subtly incompatible normalization between what R does and what Python would have done, so `path.expand()` feels a bit safer.

Reply:
But since only R does tilde expansion by default we of course want to do it before passing any paths to Python. :-)

Reply:
Great!
Good to learn the difference between both functions. I have already modified the pull request code.

Just out of curiosity, in the conda.R file, around line 258, there are several `path.expand` with absolute paths (hard-coded) as input, would this have any effect on other OS or architecture?

```r
      path.expand("/anaconda/bin/conda"),
      path.expand("/anaconda2/bin/conda"),
      path.expand("/anaconda3/bin/conda"),
      path.expand("/anaconda4/bin/conda"),
      path.expand("/miniconda/bin/conda"),
      path.expand("/miniconda2/bin/conda"),
      path.expand("/miniconda3/bin/conda"),
      path.expand("/miniconda4/bin/conda")

Reply:
Those calls to `path.expand()` are effectively un-needed / superfluous; I'm guessing they're there just for symmetry with other parts of the code. They won't have any effect on the passed-in path.